magicmetamodel/src/main/kotlin/org/jetbrains/magicmetamodel/impl/workspacemodel/impl/updaters/transformers/ModuleDetailsToPythonModuleTransformer.kt

```diff
@@ -0,0 +1,71 @@
+package org.jetbrains.magicmetamodel.impl.workspacemodel.impl.updaters.transformers
+
+import ch.epfl.scala.bsp4j.BuildTarget
+import ch.epfl.scala.bsp4j.BuildTargetDataKind
+import ch.epfl.scala.bsp4j.PythonBuildTarget
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import org.jetbrains.magicmetamodel.ModuleNameProvider
+import org.jetbrains.magicmetamodel.impl.workspacemodel.ModuleDetails
+import org.jetbrains.magicmetamodel.impl.workspacemodel.impl.updaters.Module
+import org.jetbrains.magicmetamodel.impl.workspacemodel.impl.updaters.PythonModule
+import org.jetbrains.magicmetamodel.impl.workspacemodel.impl.updaters.PythonSdkInfo
+import java.nio.file.Path
+import kotlin.io.path.Path
+
+internal class ModuleDetailsToPythonModuleTransformer(
+  moduleNameProvider: ModuleNameProvider,
+  private val projectBasePath: Path,
+) : ModuleDetailsToModuleTransformer<PythonModule>(moduleNameProvider) {
+
+  override val type = "PYTHON_MODULE"
+
+  private val sourcesItemToPythonSourceRootTransformer = SourcesItemToPythonSourceRootTransformer(projectBasePath)
+  private val resourcesItemToPythonResourceRootTransformer =
+    ResourcesItemToPythonResourceRootTransformer(projectBasePath)
+
+  override fun transform(inputEntity: ModuleDetails): PythonModule =
+    PythonModule(
+      module = toModule(inputEntity),
+      sourceRoots = sourcesItemToPythonSourceRootTransformer.transform(inputEntity.sources.map {
+        BuildTargetAndSourceItem(
+          inputEntity.target,
+          it,
+        )
+      }),
+      resourceRoots = resourcesItemToPythonResourceRootTransformer.transform(inputEntity.resources),
+      libraries = DependencySourcesItemToPythonLibraryTransformer.transform(inputEntity.dependenciesSources),
+      sdkInfo = toSdkInfo(inputEntity)
+    )
+
+  override fun toModule(inputEntity: ModuleDetails): Module {
+    val bspModuleDetails = BspModuleDetails(
+      target = inputEntity.target,
+      allTargetsIds = inputEntity.allTargetsIds,
+      dependencySources = inputEntity.dependenciesSources,
+      type = type,
+      javacOptions = null,
+      pythonOptions = inputEntity.pythonOptions,
+    )
+
+    return bspModuleDetailsToModuleTransformer.transform(bspModuleDetails)
+  }
+
+  private fun toSdkInfo(inputEntity: ModuleDetails): PythonSdkInfo? {
+    val pythonBuildTarget = extractPythonBuildTarget(inputEntity.target)
+    return if (pythonBuildTarget != null && pythonBuildTarget.version != null && pythonBuildTarget.interpreter != null)
+      PythonSdkInfo(
+        version = pythonBuildTarget.version,
```

Comment:
now u are not using `interpreter` leter - it's be nice to put here just full name of the sdk since it's the very last moment when u have a 'real' target name (so the same one as the one in collect project)